### PR TITLE
Publishes snapshot to maven and adds DEV guide

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Publish snapshot to maven and added dev guide to the repo.

Once https://github.com/opensearch-project/opensearch-build/issues/4106 is done, we can open this PR.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/72

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — COMMENTED** [review]
Please also add something to the developer guide telling people how to use this snapshot to install the extension.  Does it publish a .zip?

**Comment by owaiskazi19** [comment]
```
snapshots/
└── org
    └── opensearch
        └── plugin
            └── opensearch-ai-flow-framework
                ├── 3.0.0.0-SNAPSHOT
                │   ├── maven-metadata.xml
                │   ├── maven-metadata.xml.md5
                │   ├── maven-metadata.xml.sha1
                │   ├── maven-metadata.xml.sha256
                │   ├── maven-metadata.xml.sha512
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.pom
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.pom.md5
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.pom.sha1
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.pom.sha256
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.pom.sha512
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.zip
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.zip.md5
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.zip.sha1
                │   ├── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.zip.sha256
                │   └── opensearch-ai-flow-framework-3.0.0.0-20231005.170838-1.zip.sha512
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                ├── maven-metadata.xml.sha1
                ├── maven-metadata.xml.sha256
                └── maven-metadata.xml.sha512

**Review by dbwiddis — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+3. Launch Intellij IDEA, Choose Import Project and select the settings.gradle file in the root of this package.
+
+### Build
+
+This package uses the [Gradle](https://docs.gradle.org/current/userguide/userguide.html) build system. Gradle comes with excellent documentation that should be your first stop when trying to figure out how to operate or modify the build. we also use the OpenSearch build tools for Gradle. These tools are idiosyncratic and don't always follow the conventions and instructions for building regular Java code using Gradle. Not everything in this package will work the way it's described in the Gradle documentation. If you encounter such a situation, the OpenSearch build tools [source code](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/main/groovy/org/opensearch/gradle) is your best bet for figuring out what's going on.
+
+#### Building from the command line
+
+1. `./gradlew check` builds and tests.
+2. `./gradlew :run` launches a single node cluster with anomaly-detection (and job-scheduler) plugin installed.
```
Comment (joshpalis) [inline DEVELOPER_GUIDE.md:39]: lets remove this description

**Review by joshpalis — APPROVED** [review]
Approved with a small comment

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (owaiskazi19) [reply DEVELOPER_GUIDE.md]: Great catch! Pushed the change

**Comment by owaiskazi19** [comment]
Will merge this after https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/52

**Pull request merged**